src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -634,6 +641,51 @@ def ok_to_stop(self,
             return HandleCommandResult(-errno.EBUSY, '', warn_message)
         return HandleCommandResult(0, warn_message, '')
 
+    def find_prometheus_credentials(self, multicluster_config: Dict[str, Any], url: str) -> Optional[Dict[str, Any]]:
+        for _, clusters in multicluster_config['config'].items():
+            for cluster in clusters:
+                prometheus_url = cluster.get('prometheus_url')
+                if prometheus_url:
+                    valid_url = prometheus_url.replace("https://", "").replace("http://", "")  # since target URLs are without scheme
+                    if valid_url == url:  # check if the target URL matches with the prometheus URL (without scheme) in the config
+                        return cluster.get('prometheus_access_info')
+        return None
+    
+    def get_cluster_credentials(self, targets: List[str]) -> Dict[str, Any]:
+        clusters_credentials = {}
+        multi_cluster_config_str = str(self.mgr.get_module_option_ex('dashboard', 'MULTICLUSTER_CONFIG'))
+        try:
+            multi_cluster_config = json.loads(multi_cluster_config_str)
+        except json.JSONDecodeError as e:
+            multi_cluster_config = None
+            logger.error(f'Invalid JSON format for multi-cluster config: {e}')
+
+        if multi_cluster_config:
+            for url in targets:
+                credentials = self.find_prometheus_credentials(multi_cluster_config, url)
+                if credentials:
+                    clusters_credentials[url] = credentials
+                    clusters_credentials[url]['cert_file_name'] = ''
+        return clusters_credentials
+    
+    def generate_cert_filenames_for_clusters(self, targets: List[str]) -> Tuple[Dict[str, Any], Dict[str, Any]]:
+        r2: Dict[str, Any] = {'files': {}}
+        clusters_credentials = self.get_cluster_credentials(targets=targets)
+        unique_id_counter = 1
+        for url, credentials in clusters_credentials.items():
+            unique_id = unique_id_counter
+            unique_id_counter += 1
+            r2['files'][f'prometheus_{unique_id}_cert.crt'] = credentials['certificate']
+            credentials['cert_file_name'] = f'prometheus_{unique_id}_cert.crt'
+        return r2, clusters_credentials
+    
+    def check_target_cluster_federate_path(self, targets: List[str]) -> str:
```

Comment:
the name of this function is misleading: it's not checking anything, it's getting the `path` .. can you plz fix its name?